label_studio/data_import/functions.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import traceback
+import time
+
+from django.db import transaction
+
+from projects.models import ProjectImport
+from .serializers import ImportApiSerializer
+from .uploader import load_tasks_for_async_import
+from users.models import User
+from webhooks.utils import emit_webhooks_for_instance
+from webhooks.models import WebhookAction
+
+logger = logging.getLogger(__name__)
+
+
+def async_import_background(import_id, user_id):
+    with transaction.atomic():
+        try:
+            project_import = ProjectImport.objects.get(id=import_id)
+        except ProjectImport.DoesNotExist:
+            logger.error(f'ProjectImport with id {import_id} not found, import processing failed')
+            return
+        if project_import.status != ProjectImport.Status.CREATED:
+            logger.error(f'Processing import with id {import_id} already started')
+            return
+        project_import.status = ProjectImport.Status.IN_PROGRESS
+        project_import.save(update_fields=['status'])
+
+    user = User.objects.get(id=user_id)
+
+    start = time.time()
+    project = project_import.project
+    tasks = None
+    # upload files from request, and parse all tasks
+    # TODO: Stop passing request to load_tasks function, make all validation before
+    parsed_data, file_upload_ids, could_be_tasks_lists, found_formats, data_columns = load_tasks_for_async_import(project_import, user)
+
+    if project_import.preannotated_from_fields:
+        # turn flat task JSONs {"column1": value, "column2": value} into {"data": {"column1"..}, "predictions": [{..."column2"}]  # noqa
+        parsed_data = reformat_predictions(parsed_data, project_import.preannotated_from_fields)
+
+    if project_import.commit_to_project:
+        # Immediately create project tasks and update project states and counters
+        serializer = ImportApiSerializer(data=tasks, many=True)
+        serializer.is_valid(raise_exception=True)
+        tasks = serializer.save(project_id=project.id)
+        emit_webhooks_for_instance(user.active_organization, project, WebhookAction.TASKS_CREATED, tasks)
+
+        # return tasks, serializer
+        # tasks, serializer = self._save(parsed_data)
+        task_count = len(tasks)
+        annotation_count = len(serializer.db_annotations)
+        prediction_count = len(serializer.db_predictions)
+        # Update counters (like total_annotations) for new tasks and after bulk update tasks stats. It should be a
+        # single operation as counters affect bulk is_labeled update
+        project.update_tasks_counters_and_task_states(tasks_queryset=tasks, maximum_annotations_changed=False,
```

Comment:
Fixed. 
@makseq Do we plan to add the black linter in this repo?

Reply:
yeh, I would add it, but we can't process all files at once, because we will get lots of merge conflicts. So, I would set up black in this repo, but I would add django app one by one to it.